### DISCUSSION TITLE
SPM  doesn't show data

### DISCUSSION THREAD

**Discussion by vitalykarasik** [jaegertracing/jaeger#3761] [Q&A] [answered]
I'm trying to activate SPM based on  https://www.jaegertracing.io/docs/1.35/spm/, and getting "No data yet! ".
According to the doc, we should configure jaeger-query to use the Prometheus backend for metrics. But the doc doesn't mention jaeger-collector...
Shouldn't we point jaeger-collector to the Prometheus as well?

Thanks, Vitaly

![Screen Shot 2022-06-16 at 15 50 51](https://user-images.githubusercontent.com/7628795/174073612-1f2b5ca4-4b8b-46e9-b3c7-d537da59dafb.png)

**Comment by albertteoh** [comment, marked as answer]
The components you need are:
- OpenTelemetry Collector that will receive traces, with its pipeline configured ([example](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/spanmetricsprocessor/testdata/config-simplest.yaml)) with:
  - spanmetrics processor
  - prometheus exporter (pointing to the prometheus component below)
- Prometheus-compatible server that will receive metrics from OpenTelemetry Collector.
- Jaeger-query that will read metrics from the Prometheus-compatible server above.

You'll then need to send spans to OpenTelemetry collector and metrics should be visible in the SPM view of Jaeger UI.

Technically, jaeger-collector is not necessary at all to view metrics in SPM.

**Reply by vitalykarasik** [reply]
>OpenTelemetry Collector that will receive traces
So this is an additional component, not from Jaeger project?

thank you, Vitaly

**Reply by albertteoh** [reply]
Yes, that's correct: https://opentelemetry.io/docs/collector/getting-started/
